### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by rchl**
More details in the commit.

### CODE REVIEW

File: Default (Windows).sublime-keymap [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[
+    { "keys": ["ctrl+shift+f11"], "command": "rsync_ssh_sync_specific_remote" },
```
Comment (rchl) [inline Default (Windows).sublime-keymap]: I'm surprised that linux equivalent uses 'super' instead of 'ctrl'. I think it might not be working properly.

**Comment by rchl** [comment]
Let me rebase to get rid of fixups when (and if) you accept this patch.

**Comment by rchl** [comment]
Interestingly by default rsync changes remote file permissions for me, which is quite useless because it makes remote files innaccessible through the web server.
Google revealed that adding --perm should help but it didn't for me (in fact you already have -p in the command line). Instead --no-perm worked per this answer: http://unix.stackexchange.com/a/218165/133646
People there say it shouldn't but it did for me. Cygwin quirk, I guess.
I'm thinking it might be a good idea to add --no-perm by default. At least on Windows. Otherwise it can be both annoying and useless for people on Windows who will try the plugin.

**Comment by davidolrik** [comment]
This is good stuff - I rarely use Windows my self, but it doesn't look too complicated/interfering with how unix does stuff. 👍🏾

On unix, I always want `preserve permissions` as that seems most intuitive, so if I change a permission locally that will persist to the other side.

Maybe we should add `--chmod=ugo=rwX` as suggested [here](http://stackoverflow.com/questions/6649527/rsync-does-not-properly-set-permissions-on-windows-folder)?

**Comment by davidolrik** [comment]
Thinking about adding a general `rsync_permission_handling` option, that has sane defaults for each platform - and can be overridden if needed.

**Comment by rchl** [comment]
OK, I'll check later if your suggestion works on Windows.
Cygwin is a bit retarted when it comes file permissions (for obvious reasons - Windows file system can't  directly map to unix file system), so it's not necessairly gonna work to have local permissions map server. That is why I think it might be necessary to not carry on permission from local system on Windows for things to work reasonably well.

**Comment by davidolrik** [comment]
Then `--chmod=ugo=rwX` is what you want, it'll set the permissions according to the umask on the remote.

**Comment by rchl** [comment]
I've tried --chmod=ugo=rwX. It didn't make file inaccessible to non-owners, but it still has changed permissions.
Before: -rwxr--r-- 
After: -rwxrwxrwx
So I would probably prefer having --no-perms on Windows by default. What do you think?

**Comment by davidolrik** [comment]
What does `umask` say on the remote?

The `--chmod=ugo=rwX` option is supposed to respect that, and set the permissions accordingly.

If you create a new file on the remote, what permissions does it have?

**Comment by rchl** [comment]
Btw. This was with NO --delete specified in RSync settings and modifying an existing file.

As for the umask, it is 0022.
Creating new file:
```
echo foo > foo && ll foo
-rw-r--r-- 1 rchlodnicki rchlodnicki 4 Sep 12 09:28 foo
```

**Comment by rchl** [comment]
I've now read the thread you linked and noticed that it says to make sure "--perms" is off when using "--chmod". So I've added back "--no-perms" (in addition to --chmod=ugo=rwX) and re-checked.
 - Modififying existing file doesn't change permission.
 - Creating new file creates it with: -rwxr-xr-x (with just "--no-perms" it is "-rwxr-x---").

So I think adding --chmod=ugo=rwX is a good idea, but --no-perms would still need to be set for Windows.

**Comment by rchl** [comment]
Well, given that `--chmod` won't work without `--no-perms', it only makes sense to set it for Windows, obviously.

**Comment by davidolrik** [comment]
Totally agree about the defaults for windows. - Do the rebase, and I'll merge it in =)

I'll probably move the permission handling into a config option called `rsync_permission_options` or something like that sometime in the future, so everyone can get the permission handling they want.

**Comment by rchl** [comment]
Rebase done.
Hope we can get Windows defaults changed ASAP as in current state the plugin would be useless to me at least.

**Pull request merged**